### ISSUE TITLE
Support terraform import of existing resources

### ISSUE THREAD

**Issue by disaac** [Backblaze/terraform-provider-b2#8]
**Enhancement**
Support [importing existing resources](https://www.terraform.io/docs/extend/resources/import.html) into terraform state with provider. 

Example: 

With an existing key called testkey in your backblaze account for b2. 

```HCL
resource "b2_application_key" "testkey" {
  key_name = "testkey"
  capabilities = [
    "deleteFiles",
    "listAllBucketNames",
    "listBuckets",
    "listFiles",
    "readBuckets",
    "readFiles",
    "shareFiles",
    "writeFiles"
  ]
}
data "b2_application_key" "testkey" {
  key_name = b2_application_key.testkey.key_name
}
```
```bash
terraform import b2_application_key.testkey testkey
```

**Comment by Hjdskes** [comment]
Hey folks! I'm having a go at this since I need this for a project of mine. I followed https://learn.hashicorp.com/tutorials/terraform/provider-import?in=terraform/providers and did essentially the bare minimum that might make this work, which is to add an `ImportStatePassthroughContext` to proxy the `Read` function. I ran `make deps && make build && make install`, which installed the plugin into `~/.terraform.d/plugins/localhost/backblaze/b2/0.7.1/linux_amd64/terraform-provider-b2`.

Here's the diff to be explicit:

```
diff --git b2/resource_b2_application_key.go b2/resource_b2_application_key.go
index e8dc049..aba7a22 100644
--- b2/resource_b2_application_key.go
+++ b2/resource_b2_application_key.go
@@ -25,6 +25,12 @@ func resourceB2ApplicationKey() *schema.Resource {
                CreateContext: resourceB2ApplicationKeyCreate,
                ReadContext:   resourceB2ApplicationKeyRead,
                DeleteContext: resourceB2ApplicationKeyDelete,
+               Importer: &schema.ResourceImporter{
+                       StateContext: schema.ImportStatePassthroughContext,
+               },
 
                Schema: map[string]*schema.Schema{
                        "capabilities": {
```

I then updated my Terraform config to use this version, like so:

```
terraform {
  required_version = ">= 1.0.0"
  
  required_providers {
    b2 = {
      source  = "localhost/backblaze/b2"
      version = "~> 0.7.1"
    }
  }
}
```

However, when I try to import e.g. an application key, I get

```
 > terraform import module.backblaze.b2_application_key.foo <ID as found on the webpage>
module.backblaze.b2_application_key.rfoo: Importing from ID "REDACTED"...
module.backblaze.b2_application_key.foo: Import prepared!
  Prepared b2_application_key for import
module.backblaze.b2_application_key.foo: Refreshing state... [id=REDACTED]
╷
│ Error: Traceback (most recent call last):
│   File "b2_terraform/provider_tool.py", line 516, in <module>
│   File "b2_terraform/provider_tool.py", line 510, in main
│   File "b2_terraform/api_wrapper.py", line 19, in __init__
│   File "b2sdk/account_info/in_memory.py", line 42, in __init__
│   File "b2sdk/account_info/upload_url_pool.py", line 83, in __init__
│   File "b2sdk/account_info/upload_url_pool.py", line 91, in _reset_upload_pools
│   File "logfury/v0_1/trace_call.py", line 84, in wrapper
│ TypeError: UploadUrlPool.__init__() takes 1 positional argument but 2 were given
│ [159648] Failed to execute script 'provider_tool' due to unhandled exception!
│ 
│ 
│ 
╵
```

I also tried to use a dedicated function instead of `ImportStatePassthroughContext` (which I can confirm is being used through my custom error message being printed) but I run into the same Python error as above. I can share this code if you think it's relevant.

I've had a quick peek at the Python SDK but I couldn't identify the problem. I'm not super familiar with Python nor the B2 SDK implementation though, so I may have overlooked something. Can you help me out? :)

**Comment by dcardellino** [comment]
Any update on this topic?

**Comment by ppolewicz** [comment]
This was resolved by #32
